testdata/p4_16_samples/parser-non-field-lhs-bmv2.stf

```diff
@@ -0,0 +1,2 @@
+packet 0  DEADBEEF DEADBEEF
+expect 0  00010203 DEADBEEF   
```

Comment:
The output packet should be `00010203 DEADBEEF DEADBEEF`, yes?

If so, this test would be more strict and well-documented if you changed the `expect` line to this:
```
expect 0 00010203 DEADBEEF DEADBEEF $
```
The `$` at the end of an `expect` line in an STF test means "and that should be the end of the output packet".  Without it, `expect` lines only check that the packet begins as specified, but can have any additional data following that.

Reply:
You are right! My understanding of how this program should behave was wrong. And I learned a bit more STF here.